gobblin-service/src/main/java/org/apache/gobblin/service/monitoring/GitFlowGraphMonitor.java

```diff
@@ -0,0 +1,134 @@
+ * <root_flowGraph_dir>/<nodeName>/<nodeName>.properties
+ * <root_flowGraph_dir>/<nodeName1>/<nodeName2>/<edgeName>.properties
+ */
+@Slf4j
+public class GitFlowGraphMonitor extends GitMonitoringService implements FlowGraphMonitor {
+  public static final String GIT_FLOWGRAPH_MONITOR_PREFIX = "gobblin.service.gitFlowGraphMonitor";
+  private static final String PROPERTIES_EXTENSIONS = "properties";
+  private static final String CONF_EXTENSIONS = "conf";
+  private static final String DEFAULT_GIT_FLOWGRAPH_MONITOR_REPO_DIR = "git-flowgraph";
+  private static final String DEFAULT_GIT_FLOWGRAPH_MONITOR_FLOWGRAPH_DIR = "gobblin-flowgraph";
+  private static final String DEFAULT_GIT_FLOWGRAPH_MONITOR_BRANCH_NAME = "master";
+  static final String SHOULD_CHECKPOINT_HASHES = "shouldCheckpointHashes";
+
+  private static final int DEFAULT_GIT_FLOWGRAPH_MONITOR_POLLING_INTERVAL = 60;
+
+  private static final Config DEFAULT_FALLBACK = ConfigFactory.parseMap(ImmutableMap.<String, Object>builder()
+      .put(ConfigurationKeys.GIT_MONITOR_REPO_DIR, DEFAULT_GIT_FLOWGRAPH_MONITOR_REPO_DIR)
+      .put(ConfigurationKeys.GIT_MONITOR_CONFIG_BASE_DIR, DEFAULT_GIT_FLOWGRAPH_MONITOR_FLOWGRAPH_DIR)
+      .put(ConfigurationKeys.GIT_MONITOR_BRANCH_NAME, DEFAULT_GIT_FLOWGRAPH_MONITOR_BRANCH_NAME)
+      .put(ConfigurationKeys.GIT_MONITOR_POLLING_INTERVAL, DEFAULT_GIT_FLOWGRAPH_MONITOR_POLLING_INTERVAL)
+      .put(ConfigurationKeys.JAVA_PROPS_EXTENSIONS, PROPERTIES_EXTENSIONS)
+      .put(ConfigurationKeys.HOCON_FILE_EXTENSIONS, CONF_EXTENSIONS)
+      .put(SHOULD_CHECKPOINT_HASHES, false)
+      .build());
+
+  private Optional<? extends FSFlowTemplateCatalog> flowTemplateCatalog;
+  private final CountDownLatch initComplete;
+
+  public GitFlowGraphMonitor(Config config, Optional<? extends FSFlowTemplateCatalog> flowTemplateCatalog,
+      FlowGraph graph, Map<URI, TopologySpec> topologySpecMap, CountDownLatch initComplete) {
+    super(config.getConfig(GIT_FLOWGRAPH_MONITOR_PREFIX).withFallback(DEFAULT_FALLBACK));
+    Config configWithFallbacks = config.getConfig(GIT_FLOWGRAPH_MONITOR_PREFIX).withFallback(DEFAULT_FALLBACK);
+    this.flowTemplateCatalog = flowTemplateCatalog;
+    this.initComplete = initComplete;
+    this.listeners.add(new GitFlowGraphListener(
+        flowTemplateCatalog, graph, topologySpecMap, configWithFallbacks.getString(ConfigurationKeys.GIT_MONITOR_REPO_DIR),
+        configWithFallbacks.getString(ConfigurationKeys.GIT_MONITOR_CONFIG_BASE_DIR), configWithFallbacks.getString(ConfigurationKeys.JAVA_PROPS_EXTENSIONS),
+        configWithFallbacks.getString(ConfigurationKeys.HOCON_FILE_EXTENSIONS))
+    );
+  }
+
+  /**
+   * Determine if the service should poll Git. Current behavior is both leaders and followers(s) will poll Git for
+   * changes to {@link FlowGraph}.
+   */
+  @Override
+  public boolean shouldPollGit() {
+    return this.isActive;
```

Comment:
code says it should poll git only when it is active, but the javadoc says both the leader and followers should poll the git. Do the follower also have isActive=true?

Reply:
Good callout, so followers won't have `isActive=true`, but for followers they need to also set the flow catalog and the specCompiler as active too, so I'll leave it as this for now to maintain previous behavior, but for multi master we need some changes in `GobblinServiceManager` to handle enabling the flowGraph plus the catalog and spec compiler